CPO-583 Reduce output from Terraform

tom-mckay opened the pull request:
Add the procore_terraform_wrapper.py script to reduce the output from Terraform and fix an issue with return codes not passing through tfmask.

https://procoretech.atlassian.net/browse/CPO-583

tom-mckay committed:
add procore_terraform_wrapper

tom-mckay committed:
remove unnecessary name in logging format

rarkebauer reviewed (changes requested):
I don't understand the README.bak.md and blank README.md. Did you mean to leave the README.md as is in this PR?

Can you also include a short readme in the procore_terraform_wrapper with the commands to run the script and the tests?

tom-mckay commented:
> I don't understand the README.bak.md and blank README.md. Did you mean to leave the README.md as is in this PR?
> 
> Can you also include a short readme in the procore_terraform_wrapper with the commands to run the script and the tests?

I moved the upstream's README.md to README.bak.md and replaced README.md with content specific to our fork of the upstream. It's not blank.

rarkebauer commented:
whoops I didn't realize github was hiding it. Can you still add a note about running/testing the procore_terraform_wrapper locally?

brandonpollard commented on .vscode/settings.json:
```diff
@@ -0,0 +1,10 @@
+{
```
Shouldn't this be ignored?

tom-mckay replied:
Not necessarily. This file is helpful for vscode users as it sets up the linters to run automatically.

tom-mckay committed:
update README.md

tom-mckay committed:
add a few more comments

rarkebauer commented on README.md:
````diff
@@ -1,169 +1,129 @@
+Run the tests after any changes to the python code.
+
+```
+pip3 install -f requirements.txt
````
```suggestion
pip3 install requirements.txt
```

tom-mckay replied:
This was supposed to be `-r`. Fixed.

rarkebauer commented on README.md:
````diff
@@ -1,169 +1,129 @@
-
-or use --repo-config-json='{"repos":[{"id":"/.*/", "apply_requirements":["approved", "mergeable"], "allowed_overrides":["apply_requirements","workflow"], "allow_custom_workflows":true}]}'
 ```
+usage: procore_terraform_wrapper.py [-h] --action {init,plan,apply}
````
```suggestion
usage: ./procore_terraform_wrapper.py [-h] --action {init,plan,apply}
```

tom-mckay replied:
I generated this by copy-pasting the output of `./procore_terraform_wrapper.py -h`. The name of the script is automatically generated by argparse. I'd rather not change it manually in the readme.

rarkebauer replied:
I would prefer to make the readme the source of copy/pasteable commands. I think it is easier when someone is completely unfamiliar with the repo. But that doesn't need to be a blocker if we disagree

tom-mckay committed:
redirect stderr to stdout

tom-mckay committed:
Add support for extra args with COMMENT_ARGS environment variable

tom-mckay committed:
fix pip command

tom-mckay commented:
I'm going to hold off on merging this until we have the staging atlantis bot and repo ready.

tom-mckay committed:
fix incorrect path in circleci config

tom-mckay committed:
Merge branch 'main' into cpo-583-reduce-output

tom-mckay committed:
add debugging to atlantis staging

tom-mckay committed:
debugging

tom-mckay committed:
rework args

tom-mckay committed:
new temp image

tom-mckay committed:
test leaving comment_args unquoted

tom-mckay committed:
wrap up updates

tom-mckay commented:
Ok, this is ready for re-review. I fixed issues that came up during testing:
- Use arguments instead of environment variables to make it easier to test.
- Use `shell=True` to fix strange issues when run via Atlantis

A PR is available to test out the new code here: procore/atlantis-test-repository#3
